### PR TITLE
name ML properties in the results file by architecture, energy become…

### PR DESCRIPTION
**Pull request by alinelena**
…s mace_energy,...

- [x] tests

### CODE REVIEW

**Review by ElliottKasoar — COMMENTED** [review]
At the moment this seems to write out things twice:

```
Lattice="5.62 0.0 0.0 0.0 5.62 0.0 0.0 0.09808252417753258 5.619144046778919" 
Properties=species:S:1:pos:R:3:spacegroup_kinds:I:1:mace_mp_forces:R:3:forces:R:3
spacegroup="P 1" 
unit_cell=conventional
occupancy="_JSON {\"0\": {\"Na\": 1.0}, \"1\": {\"Cl\": 1.0}, \"2\": {\"Na\": 1.0}, \"3\": {\"Cl\": 1.0}, \"4\": {\"Na\": 1.0}, \"5\": {\"Cl\": 1.0}, \"6\": {\"Na\": 1.0}, \"7\": {\"Cl\": 1.0}}" 
mace_mp_energy=-27.020331256158105
mace_mp_stress="-0.006830042553032722 -0.006875546081961216 -0.006926724236709431 0.0012606249302715168 -1.3664263946131713e-19 3.979321394392908e-19"
energy=-27.020331256158105 
stress="-0.006830042553032722 3.979321394392908e-19 -1.3664263946131713e-19 3.979321394392908e-19 -0.006875546081961216 0.0012606249302715168 -1.3664263946131713e-19 0.0012606249302715168 -0.006926724236709431"
free_energy=-27.020331256158105
pbc="T T T"
```

Looks mostly ok otherwise

**Comment by alinelena** [comment]
> At the moment this seems to write out things twice:
> 
> ```
> Lattice="5.62 0.0 0.0 0.0 5.62 0.0 0.0 0.09808252417753258 5.619144046778919" 
> Properties=species:S:1:pos:R:3:spacegroup_kinds:I:1:mace_mp_forces:R:3:forces:R:3
> spacegroup="P 1" 
> unit_cell=conventional
> occupancy="_JSON {\"0\": {\"Na\": 1.0}, \"1\": {\"Cl\": 1.0}, \"2\": {\"Na\": 1.0}, \"3\": {\"Cl\": 1.0}, \"4\": {\"Na\": 1.0}, \"5\": {\"Cl\": 1.0}, \"6\": {\"Na\": 1.0}, \"7\": {\"Cl\": 1.0}}" 
> mace_mp_energy=-27.020331256158105
> mace_mp_stress="-0.006830042553032722 -0.006875546081961216 -0.006926724236709431 0.0012606249302715168 -1.3664263946131713e-19 3.979321394392908e-19"
> energy=-27.020331256158105 
> stress="-0.006830042553032722 3.979321394392908e-19 -1.3664263946131713e-19 3.979321394392908e-19 -0.006875546081961216 0.0012606249302715168 -1.3664263946131713e-19 0.0012606249302715168 -0.006926724236709431"
> free_energy=-27.020331256158105
> pbc="T T T"
> ```
> 
> Looks mostly ok otherwise

yes on it to fix it but is more delicate than initially thought... by time you are in london shall be done.

**Review by ElliottKasoar — COMMENTED** [review]
Actually, we might want to write the info from the `results` dictionary, rather than via the calculate functions, so we clean the NaNs etc.

**Comment by ElliottKasoar** [comment]
It looks like you've already done the equivalent changes for the Python tests, but `test_singlepoint_cli.py::test_singlepoint` fails because now that we don't write the results of the calculator directly, it no longer attaches a `SinglePointCalculator` when the data is read back in, and so `get_potential_energy` etc. can't be called.

Is this definitely what we want?

Might it be preferable to have some sort of store-as-info setting, rather than the default? Otherwise, it feels hidden away a little.

It also means if we do something like `single_point.run()` etc. repeatedly, it'll have to explicitly recalculate everything, when it might already be stored, just not where it expects.

**Comment by alinelena** [comment]
just to note, this is an abuse of ase atoms object. till a proper mechanism to store user properties will be available

File: janus_core/calculations/single_point.py [hunk 1]
```diff
@@ -216,10 +216,16 @@ def _get_potential_energy(self) -> MaybeList[float]:
         MaybeList[float]
             Potential energy of structure(s).
         """
+        tag = f"{self.architecture}_energy"
         if isinstance(self.struct, list):
-            return [struct.get_potential_energy() for struct in self.struct]
+            energies = [struct.get_potential_energy() for struct in self.struct]
+            for energy, struct in zip(energies, self.struct):
+                struct.info[tag] = energy
```
Comment (oerc0122) [inline janus_core/calculations/single_point.py:223]: Really finicky thing and purely stylistic, but I think that it's best to keep things in same positions as much as possible through the code where possible.
```suggestion
            for struct, energy  in zip(self.struct, energies):
                struct.info[tag] = energy
```

File: janus_core/calculations/single_point.py [hunk 2]
```diff
@@ -230,10 +236,16 @@ def _get_forces(self) -> MaybeList[ndarray]:
         MaybeList[ndarray]
             Forces of structure(s).
         """
+        tag = f"{self.architecture}_forces"
         if isinstance(self.struct, list):
-            return [struct.get_forces() for struct in self.struct]
+            forces = [struct.get_forces() for struct in self.struct]
+            for force, struct in zip(forces, self.struct):
+                struct.arrays[tag] = force
```
Comment (oerc0122) [inline janus_core/calculations/single_point.py:243]:
```suggestion
            for struct, force in zip(self.struct, forces):
                struct.arrays[tag] = force
```

File: janus_core/calculations/single_point.py [hunk 3]
```diff
@@ -244,13 +256,19 @@ def _get_stress(self) -> MaybeList[ndarray]:
         MaybeList[ndarray]
             Stress of structure(s).
         """
+        tag = f"{self.architecture}_stress"
         if isinstance(self.struct, list):
-            return [struct.get_stress() for struct in self.struct]
+            stresses = [struct.get_stress() for struct in self.struct]
+            for stress, struct in zip(stresses, self.struct):
+                struct.info[tag] = stress
```
Comment (oerc0122) [inline janus_core/calculations/single_point.py:263]:
```suggestion
            for struct, stress in zip(self.struct, stresses):
                struct.info[tag] = stress
```

**Review by oerc0122 — APPROVED** [review]
Apart from one stylistic thing, if this works, I think it's fine.

**Review by oerc0122 — DISMISSED** [review]

**Review by oerc0122 — APPROVED** [review]

**Pull request merged**